src/v/archival/ntp_archiver_service.cc

```diff
@@ -339,13 +339,13 @@ ntp_archiver::upload_segment(upload_candidate candidate) {
       "current term: {}, "
       "original term: {}",
       [this, original_term](cloud_storage::lazy_abort_source& las) {
-          auto lost_leadership = !_partition->is_leader()
+          auto lost_leadership = !_partition->is_elected_leader()
```

Comment:
Do I understand that the change was made because when `is_leader` returns `false` but `is_elected_leader` is still returns 'true' we could cancel an upload?

Reply:
Yes, from what I can see, basically what happened was that during the time that the PR for leadership check #4642  was in review, the semantics of `is_leader` in the tree were changed and ntp archival (and many other places in code base) switched to `is_elected_leader`. But since the PR was in review it was not part of the refactor and so it remained on `is_leader` and was merged in that state.

Now the archivers use `is_elected_leader` but the check uses `is_leader`, causing incorrect segment upload cancellation unless the node uploading has propagated the config batch. In the check we should use the weaker query to be consistent with the rest of the system.